### PR TITLE
[plugin-web-app] Add support of CKEditor 5 text forms

### PR DESCRIPTION
**Pull request by vkepin**

### CODE REVIEW

File: vividus-plugin-web-app/src/main/java/org/vividus/steps/ui/web/FieldSteps.java [hunk 1]
```diff
@@ -154,10 +154,15 @@ private void enterTextInField(WebElement element, String text, boolean retry, Ru
...
             {
-                javascriptActions.executeScript("var element = arguments[0];element.innerHTML = arguments[1];", element,
-                        text);
+                // Workaround for CKEditor 5 (JavaScript WYSIWYG editor):
+                // https://github.com/ckeditor/ckeditor5/issues/6554
+                boolean isCKEditor = javascriptActions.executeScript(
+                        "return arguments[0].ckeditorInstance !== undefined", element);
+                String textTypeScript = isCKEditor ? "arguments[0].ckeditorInstance.setData(arguments[1])"
+                        : "arguments[0].innerHTML = arguments[1]";
+                javascriptActions.executeScript(textTypeScript, element, text);
```
Comment (uarlouski) [inline vividus-plugin-web-app/src/main/java/org/vividus/steps/ui/web/FieldSteps.java:165]: why not to write it as one execute script?

File: vividus-plugin-web-app/src/main/java/org/vividus/steps/ui/web/FieldSteps.java [hunk 2]
```diff
@@ -154,10 +154,13 @@ private void enterTextInField(WebElement element, String text, boolean retry, Ru
...
-            if (webDriverManager.isBrowserAnyOf(Browser.SAFARI) && fieldActions.isElementContenteditable(element))
+            if (fieldActions.isElementContenteditable(element))
             {
-                javascriptActions.executeScript("var element = arguments[0];element.innerHTML = arguments[1];", element,
-                        text);
+                // Workaround for CKEditor 5 (JavaScript WYSIWYG editor):
+                // https://github.com/ckeditor/ckeditor5/issues/6554
+                javascriptActions.executeScript(
+                        "arguments[0].ckeditorInstance ? arguments[0].ckeditorInstance.setData(arguments[1])"
+                                + " : arguments[0].innerHTML = arguments[1]", element, text);
```
Comment (valfirst) [inline vividus-plugin-web-app/src/main/java/org/vividus/steps/ui/web/FieldSteps.java:163]: what about `org.vividus.ui.web.action.FieldActions#addText`?

File: vividus-plugin-web-app/src/main/java/org/vividus/steps/ui/web/FieldSteps.java [hunk 3]
```diff
@@ -154,10 +154,13 @@ private void enterTextInField(WebElement element, String text, boolean retry, Ru
         {
             element.clear();
             LOGGER.info("Entering text \"{}\" in element", text);
-            if (webDriverManager.isBrowserAnyOf(Browser.SAFARI) && fieldActions.isElementContenteditable(element))
+            if (fieldActions.isElementContenteditable(element))
```
Comment (valfirst) [inline vividus-plugin-web-app/src/main/java/org/vividus/steps/ui/web/FieldSteps.java:157]: I think this logic should be moved to one class: either `FieldSteps.java` or `FieldActions`

File: vividus-plugin-web-app/src/main/java/org/vividus/ui/web/action/FieldActions.java [hunk 4]
```diff
@@ -156,6 +243,11 @@ public void typeText(WebElement element, String text)
         }
     }
 
+    private boolean isValueNotEqualTo(WebElement element, String expectedValue)
+    {
+        return !expectedValue.equals(javascriptActions.executeScript("return arguments[0].value;", element));
+    }
+
     @Override
     public boolean isElementContenteditable(WebElement element)
```
Comment (vkepin) [inline vividus-plugin-web-app/src/main/java/org/vividus/ui/web/action/FieldActions.java:252]: remove from IFieldActions

**Review by valfirst — APPROVED** [review]

**Pull request merged**